src/main/host/memory_manager.rs

```diff
@@ -329,15 +343,9 @@ where
         }
     }
 
-    fn copy(&self, offset: usize, buf: &mut [T]) -> Result<(), Errno> {
-        let nread = self
-            .memory_manager
-            .readv_ptrs(&mut [buf], &[self.ptr.slice(offset..)])?;
-        if nread != buf.len() {
-            Err(Errno::EFAULT)
-        } else {
-            Ok(())
-        }
+    fn read_some(&self, offset: usize, buf: &mut [T]) -> Result<usize, Errno> {
+        self.memory_manager
+            .readv_ptrs(&mut [buf], &[self.ptr.slice(offset..)])
     }
```

Comment:
In order to stay consistent with the other functions, I think `read_some()` should return the number of `T` items, not the number of bytes returned by `readv_ptrs()`. So it should return `self.memory_manager.readv_ptrs() / size_of(T)`.

Reply:
Done